nearswap/tests/swap.rs

```diff
@@ -0,0 +1,186 @@
+use std::convert::TryFrom;
+
+use near_sdk::json_types::{ValidAccountId, U128};
+use near_sdk::AccountId;
+use near_sdk_sim::{call, deploy, init_simulator, to_yocto, view, ContractAccount, UserAccount};
+
+use near_sdk_sim::transaction::ExecutionStatus;
+use nearswap::{NearSwapContract, PoolInfo};
+use std::collections::HashMap;
+use sample_token::ContractContract as SampleToken;
+
+mod nep141_utils;
+use nep141_utils::*;
+
+near_sdk_sim::lazy_static_include::lazy_static_include_bytes! {
+    NEARSWAP_WASM_BYTES => "../res/nearswap.wasm",
+}
+
+#[test]
+fn swap_test() {
+    let root = init_simulator(None);
+    let owner = root.create_user("owner".to_string(), to_yocto("100000"));
+    let nearswap = deploy!(
+        contract: NearSwapContract,
+        contract_id: clp_contract(),
+        bytes: &NEARSWAP_WASM_BYTES,
+        signer_account: owner,
+        init_method: new(to_va("owner".to_string()))
+    );
+    let token1 = sample_token(&owner, dai(), vec![clp_contract()]);
+    let token2 = sample_token(&owner, eth(), vec![clp_contract()]);
+    call!(
+        owner,
+        nearswap.extend_whitelisted_tokens(vec![to_va(dai()), to_va(eth())])
+    );
+    call!(
+        owner,
+        nearswap.create_pool(to_va("dai".into())),
+        deposit = to_yocto("1")
+    )
+    .assert_success();
+    call!(
+        owner,
+        nearswap.create_pool(to_va("eth".into())),
+        deposit = to_yocto("1")
+    )
+    .assert_success();
```

Comment:
let's create a common function for preparation, or some common calls, so the reading will be easier.

Reply:
ok

Reply:
added common functions for account registration and near deposits